src/java.base/share/classes/java/lang/Double.java

```diff
@@ -200,6 +200,139 @@
+ * densities of binary and decimal values at different points along the
+ * real number line).
+ *
+ * <p>This representation hazard of decimal fractions is one reason to
+ * use caution when storing monetary values as {@code float} or {@code
+ * double}. Alternatives include using an integer type and storing
+ * cents or mills or using {@link java.math.BigDecimal BigDecimal} to
+ * store decimal fraction values exactly.
+ *
+ * <p>For each finite floating-point value and a given floating-point
+ * type, there is a contiguous region of the real number line which
+ * maps to that value. Under the default round to nearest rounding
+ * policy (JLS {@jls 15.4}), this contiguous region for a value is
+ * typically one {@linkplain Math#ulp ulp} (unit in the last place)
+ * wide and centered around the exactly representable value. (At
+ * exponent boundaries, the region is asymmetrical and larger on the
+ * side with the larger exponent.) For example, for {@code 0.1f}, the
+ * region can be computed as follows:
+ *
+ * <br>// Numeric values listed are exact values
+ * <br>oneTenthApproxAsFloat = 0.100000001490116119384765625;
+ * <br>ulpOfoneTenthApproxAsFloat = Math.ulp(1.0f) = 7.450580596923828125E-9;
+ * <br>// Numeric range that is converted to the float closest to 0.1, _excludes_ endpoints
+ * <br>(oneTenthApproxAsFloat - &frac12;ulpOfoneTenthApproxAsFloat, oneTenthApproxAsFloat + &frac12;ulpOfoneTenthApproxAsFloat) =
+ * <br>(0.0999999977648258209228515625, 0.1000000052154064178466796875)
+ *
+ * <p>In particular, a correctly rounded decimal to binary conversion
+ * of any string representing a number in this range, say by {@link
+ * Float#valueOf(float)}, will be converted to the same value:
+ *
+ * {@snippet lang="java" :
+ * Float.valueOf("0.0999999977648258209228515625000001"); // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.099999998");                          // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.1");                                  // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.100000001490116119384765625");        // exact conversion
+ * Float.valueOf("0.100000005215406417846679687");        // rounds down to oneTenthApproxAsFloat
+ * Float.valueOf("0.100000005215406417846679687499999");  // rounds down to oneTenthApproxAsFloat
+ * }
+ *
+ * <p>An analogous range can be constructed similarly for the {@code
+ * double} type based on the exact value of {@code double}
+ * approximation to {@code 0.1d} and the numerical value of {@code
+ * Math.ulp(0.1d)} and likewise for other particular numerical values
+ * in the {@code float} and {@code double} types.
+ *
+ * <p>As seen in the above conversions, compared to the exact result
+ * the same floating-point value as a result can be:
+ * <ul>
+ * <li>greater than the exact result
+ * <li>less than the exact result
+ * <li>equal to the exact result
+ * </ul>
+ *
+ * A floating-point value doesn't "know" if it was the result of
```

Comment:
Would it be better to use "whether" in this listing of alternatives? (As recommended by Merriam-Webster: https://www.merriam-webster.com/grammar/if-vs-whether-difference-usage)